src/sdk/main/src/NodeCreateTransaction.cc

```diff
@@ -0,0 +1,195 @@
+  : Transaction<NodeCreateTransaction>(transactionBody)
+{
+  initFromSourceTransactionBody();
+}
+
+//-----
+NodeCreateTransaction::NodeCreateTransaction(
+  const std::map<TransactionId, std::map<AccountId, proto::Transaction>>& transactions)
+  : Transaction<NodeCreateTransaction>(transactions)
+{
+  initFromSourceTransactionBody();
+}
+
+//-----
+NodeCreateTransaction& NodeCreateTransaction::setAccountId(const AccountId& accountId)
+{
+  requireNotFrozen();
+  mAccountId = accountId;
+  return *this;
+}
+
+//-----
+NodeCreateTransaction& NodeCreateTransaction::setDescription(const std::optional<std::string>& description)
+{
+  requireNotFrozen();
+  mDescription = description;
+  return *this;
+}
+
+//-----
+NodeCreateTransaction& NodeCreateTransaction::setGossipEndpoints(const std::vector<Endpoint>& endpoints)
+{
+  requireNotFrozen();
+  mGossipEndpoints = endpoints;
+  return *this;
+}
+
+//-----
+NodeCreateTransaction& NodeCreateTransaction::setServiceEndpoints(const std::vector<Endpoint>& endpoints)
+{
+  requireNotFrozen();
+  mServiceEndpoints = endpoints;
+  return *this;
+}
+
+//-----
+NodeCreateTransaction& NodeCreateTransaction::setGossipCaCertificate(const std::vector<std::byte>& certificate)
+{
+  requireNotFrozen();
+  mGossipCaCertificate = certificate;
+  return *this;
+}
+
+//-----
+NodeCreateTransaction& NodeCreateTransaction::setGrpcCertificateHash(const std::vector<std::byte>& hash)
+{
+  requireNotFrozen();
+  mGrpcCertificateHash = hash;
+  return *this;
+}
+
+//-----
+NodeCreateTransaction& NodeCreateTransaction::setAdminKey(const std::shared_ptr<Key>& key)
+{
+  requireNotFrozen();
+  mAdminKey = key;
+  return *this;
+}
+
+//-----
+grpc::Status NodeCreateTransaction::submitRequest(const proto::Transaction& request,
+                                                  const std::shared_ptr<internal::Node>& node,
+                                                  const std::chrono::system_clock::time_point& deadline,
+                                                  proto::TransactionResponse* response) const
+{
+  return node->submitTransaction(proto::TransactionBody::DataCase::kNodeCreate, request, deadline, response);
+}
+
+//-----
+void NodeCreateTransaction::validateChecksums(const Client& client) const
+{
+  mAccountId.validateChecksum(client);
+}
+
+//-----
+void NodeCreateTransaction::addToBody(proto::TransactionBody& body) const
+{
+  body.set_allocated_nodecreate(build());
+}
+
+//-----
+void NodeCreateTransaction::initFromSourceTransactionBody()
+{
+  const proto::TransactionBody transactionBody = getSourceTransactionBody();
+
+  if (!transactionBody.has_nodecreate())
+  {
+    throw std::invalid_argument("Transaction body doesn't contain NodeCreate data");
+  }
+
+  const aproto::NodeCreateTransactionBody& body = transactionBody.nodecreate();
+
+  mAccountId = AccountId::fromProtobuf(body.account_id());
+
+  mDescription = body.description();
```

Comment:
Should check for `description` validity here (`if (body.has_description()) {...}`)

Reply:
There is no `body.has_description()` sadly. Otherwise would have checked.

Reply:
Hmm maybe `mDescription` shouldn't be `optional` then? It doesn't really make sense to have an `optional` object if there's no way to check in the protobuf if it's valid or not

Reply:
```
  /**
   * A short description of the node.
   * This value, if set, MUST NOT exceed 100 bytes when encoded as UTF-8.
   * This field is OPTIONAL.
   */
```
   IDK really lets leave it like that for now